lib/ddtrace/configuration/validation_proxy.rb

```diff
@@ -0,0 +1,44 @@
+require 'forwardable'
+
+module Datadog
+  module Configuration
+    # Forwards configuration settings allowed for tracing,
+    # but raises errors for access to anything else.
+    class TracingValidationProxy
```

Comment:
Minor: Should this go inside `Datadog::Tracing`?

Reply:
Yes, I plan on doing that when we migrate namespaces (it will become `Datadog::Tracing::Configuration::ValidationProxy`.)